lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableRemoteEvent.cs

```diff
@@ -72,7 +78,7 @@ public WritableRemoteEvent()
         /// <param name="reader">The reader from which to read </param>
         public void Read(IDataReader reader)
         {
-            Value = Activator.CreateInstance<T>();
+            Value = (T) _injector.ForkInjector().GetInstance(typeof(T));
```

Comment:
Yes we need different instance.

On Wed, May 13, 2015 at 11:46 AM, Julia Wang <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableRemoteEvent.cs
> <https://github.com/apache/incubator-reef/pull/181#discussion_r30262738>:
>
> > @@ -72,7 +78,7 @@ public WritableRemoteEvent()
> >          /// <param name="reader">The reader from which to read </param>
> >          public void Read(IDataReader reader)
> >          {
> > -            Value = Activator.CreateInstance<T>();
> > +            Value = (T) _injector.ForkInjector().GetInstance(typeof(T));
>
> Are you expecting different instance every time this method is called even
> T is the same?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/181/files#r30262738>.
>